elementary/monitor/api/tests/tests.py

```diff
@@ -10,66 +10,53 @@
 from elementary.monitor.api.invocations.invocations import InvocationsAPI
 from elementary.monitor.api.invocations.schema import DbtInvocationSchema
 from elementary.monitor.api.tests.schema import (
+    DbtTestResultSchema,
+    ElementaryTestResultSchema,
     InvocationSchema,
     InvocationsSchema,
     ModelUniqueIdType,
-    TestInfoSchema,
     TestMetadataSchema,
+    TestResultDBRowSchema,
     TestResultSchema,
     TestRunSchema,
     TestUniqueIdType,
     TotalsSchema,
 )
 from elementary.monitor.data_monitoring.schema import DataMonitoringReportFilter
-from elementary.utils.json_utils import try_load_json
 from elementary.utils.log import get_logger
 from elementary.utils.time import convert_utc_iso_format_to_datetime
 
 logger = get_logger(__name__)
 
 
-TEST_RESULTS = "test_results"
-TEST_RUNS = "test_runs"
-TESTS_METADATA = "tests_metadata"
-TESTS_SAMPLE_DATA = "tests_sample_data"
-INVOCATION = "invocation"
-TEST_INVOCATIONS = "test_invocations"
-
-
 class TestsAPI(APIClient):
     def __init__(self, dbt_runner: DbtRunner):
         super().__init__(dbt_runner)
         self.invocations_api = InvocationsAPI(dbt_runner)
 
-    @staticmethod
-    def get_test_sub_type_unique_id(
-        model_unique_id: str,
-        test_unique_id: str,
-        column_name: Optional[str] = None,
-        test_sub_type: Optional[str] = None,
-        **kwargs,
-    ) -> str:
-        return f"{model_unique_id}.{test_unique_id}.{column_name if column_name else 'None'}.{test_sub_type if test_sub_type else 'None'}"
-
-    def get_tests_metadata(
+    def get_all_test_results_db_rows(
         self,
         days_back: Optional[int] = 7,
-        invocation_id: str = None,
-        should_cache: bool = True,
-    ) -> List[TestMetadataSchema]:
+        invocations_per_test: int = 720,
+        metrics_sample_limit: int = 5,
+        disable_passed_test_metrics: bool = False,
+    ) -> List[TestResultDBRowSchema]:
         run_operation_response = self.dbt_runner.run_operation(
             macro_name="get_test_results",
-            macro_args=dict(days_back=days_back, invocation_id=invocation_id),
+            macro_args=dict(
+                days_back=days_back,
+                invocations_per_test=invocations_per_test,
+                metrics_sample_limit=metrics_sample_limit,
+                disable_passed_test_metrics=disable_passed_test_metrics,
+            ),
         )
-        tests_metadata = (
+        test_results = (
             json.loads(run_operation_response[0]) if run_operation_response else []
         )
-        tests_metadata = [
-            TestMetadataSchema(**test_metadata) for test_metadata in tests_metadata
+        test_results = [
```

Comment:
This should already be ordered and used rather than the `invocation_order`.

Reply:
Explained on the schema why we can't relay on the order 🙂  (rank instead of row_number)